File: src/main/java/plugin/interaction/combat/magic/lunar/teleport/BarbarianTeleportSpell.java
package plugin.interaction.combat.magic.lunar.teleport;

import org.redrune.game.global.WorldTile;
import org.redrune.game.content.plugin.combat.spell.type.TeleportSpellPlugin;

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 7/28/2017
 */
public class BarbarianTeleportSpell implements TeleportSpellPlugin {
	
	@Override
	public int levelRequired() {
		return 75;
	}
	
	@Override
	public int[] runesRequired() {
		return arguments(ASTRAL_RUNE, 2, LAW_RUNE, 2, FIRE_RUNE, 3);
	}
	
	@Override
	public WorldTile destination() {
		return new WorldTile(2544, 3572, 0);
	}
	
	@Override
	public int spellId() {
		return 22;
	}
	
	@Override
	public double exp() {
		return 77;
	}
	
	@Override
	public MagicBook book() {
		return MagicBook.LUNAR;
	}
}

File: src/main/java/plugin/interaction/combat/magic/lunar/teleport/BarbarianTeleportSpell.kt
package plugin.interaction.combat.magic.lunar.teleport

import org.redrune.game.content.plugin.combat.spell.type.TeleportSpellPlugin
import org.redrune.game.global.WorldTile
import org.redrune.utility.constants.MagicConstants
import org.redrune.utility.constants.MagicConstants.MagicBook

/**
 * @author Tyluur <itstyluur></itstyluur>@icloud.com>
 * @since 7/28/2017
 */
class BarbarianTeleportSpell : TeleportSpellPlugin {
    override fun levelRequired(): Int {
        return 75
    }

    override fun runesRequired(): IntArray {
        return arguments(MagicConstants.ASTRAL_RUNE, 2, MagicConstants.LAW_RUNE, 2, MagicConstants.FIRE_RUNE, 3)
    }

    override fun destination(): WorldTile {
        return WorldTile(2544, 3572, 0)
    }

    override fun spellId(): Int {
        return 22
    }

    override fun exp(): Double {
        return 77.0
    }

    override fun book(): MagicBook {
        return MagicBook.LUNAR
    }
}